src/diff.c

```diff
@@ -731,8 +764,12 @@ static int maybe_modified(
 		new_is_workdir)
 		nmode = (nmode & ~MODE_BITS_MASK) | (omode & MODE_BITS_MASK);
 
+	/* if one side is a conflict, mark the whole delta as conflicted */
+	if (git_index_entry_stage(oitem) || git_index_entry_stage(nitem))
+		status = GIT_DELTA_CONFLICTED;
```

Comment:
Can we get explicit `> 0` comparisons here? Relying on zero as false seems iffy here. It seems to say "if it has a stage" like with pointesr rather than "if the stage is higher than 0" (not so say I wouldn't have done this myself, but when someone else writes it... :)